src/expr/src/agg/def.rs

```diff
@@ -51,17 +51,15 @@ impl AggCall {
         let agg_kind = AggKind::from_protobuf(agg_call.get_type()?)?;
         let args = match &agg_call.get_args()[..] {
             [] => AggArgs::None,
-            [arg] if agg_kind != AggKind::StringAgg => {
-                AggArgs::Unary(DataType::from(arg.get_type()?), arg.get_index() as usize)
-            }
-            [agg_arg, extra_arg] if agg_kind == AggKind::StringAgg => AggArgs::Binary(
+            [arg] => AggArgs::Unary(DataType::from(arg.get_type()?), arg.get_index() as usize),
+            [agg_arg, extra_arg] => AggArgs::Binary(
```

Comment:
I've also changed this and extract the logic to `AggArgs::from_protobuf` in #9429, may have to resolve conflict for this file.